src/odemis/model/_futures.py

```diff
@@ -538,8 +546,20 @@ def _on_future_done(self, f):
             self.set_result(None)
 
     def _estimate_end(self):
-        start, end = self.get_progress()
-        return start + sum(self.futures[f] for f in self.futures if not f.done())
+        """
+        Estimate the end of the batch future based on the current time plus the
+        estimated time for all futures that still need to be executed.
+
+        returns: (float) Newly estimated end time based for the batch future.
+        """
+        duration = time.time()
+        for f, t in self.futures.items():
+            # only take futures into account that are not executed (finished) yet
+            if not f.done():
+                duration += t
```

Comment:
I liked the compact way of before. Actually, I would write:
`duration =  sum(t for f, t in self.futures.items() if not f.done())`

Note that duration implies that it's a difference of time between the start and the end. Maybe you could call it `time_left`.